### PR TITLE
[MIG] Migrate account_operating_unit to 11.0

### PR DESCRIPTION
**Pull request by michael-villamar**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please preserve commit history following technical method explained in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0

**Comment by richard-willdooit** [comment]
@pedrobaeza - we are not getting enough information from runbot to see why this is failing?

**Comment by pedrobaeza** [comment]
Yes, you can: just click on Details here, click on the gears, and "Full all logs". There, you see the warning that is causing this to go to red:

```
2018-04-16 01:57:47,530 168 WARNING openerp_test odoo.addons.base.ir.ir_ui_view: Error-prone use of @class in view report_financial_operating_units (): use the hasclass(*classes) function to filter elements by their classes
2018-04-16 01:57:47,555 168 WARNING openerp_test odoo.addons.base.ir.ir_ui_view: Error-prone use of @class in view report_trialbalance_operating_units (): use the hasclass(*classes) function to filter elements by their classes
```

**Comment by emagdalenaC2i** [comment]
Is this duplicate with https://github.com/OCA/operating-unit/pull/101 ?

**Comment by AaronHForgeFlow** [comment]
As @emagdalenaC2i said, the analytic_operating_unit module is already proposed to v11 at #101. However, AFAIK the account_operating_unit is not proposed in another PR so @michael-villamar can propose it here.

**Comment by meigallodixital** [comment]
We apply this PR on local, and all appears works well. But we have a doubt, there are any way to have or tag account journals by operating unit?

**Comment by AaronHForgeFlow** [comment]
@meigallodixital no, the operating unit is not in the journals, not sure if it is easy to implement that or not. Anyway,  as long as the operating unit is in the journal items the accounting is clearly separated between operating units.

@michael-villamar can you remove the analytic_operating_unit module from this PR? That one is proposed on #101 you can still propose the account_operating_unit here.

**Comment by meigallodixital** [comment]
To clarify, in our case we have installed the analytic_operating_unit from  #101 and  account_operating_unit from #114  :)

**Comment by AaronHForgeFlow** [comment]
@meigallodixital yes that's correct. :)

**Comment by meigallodixital** [comment]
Now that the analytic PR is merged into the main branch, it would be good to prepare and give a boost to this account_operating_unit module. 

No issues in the account module with the changes from this PR on my local enviroment (OCB).

**Comment by AaronHForgeFlow** [comment]
@michael-villamar can you remove the analytic_operating_unit changes from this PR? It is already merged. Thank you!

**Comment by meigallodixital** [comment]
What is pending to accept this module?

**Comment by AaronHForgeFlow** [comment]
@meigallodixital basically more reviews. I haven't had the time so far to review, but fixing the conflicts would make this easier to reviews by others.

**Comment by meigallodixital** [comment]
I think there are no conflicts in the merge with the base branch. It is more lack of tests to cover all the codecoverage if I'm not wrong ...

**Comment by AaronHForgeFlow** [comment]
No, I'm getting a message: "This branch cannot be rebased due to conflicts". Also, the commit history is not respected. @michael-villamar If you agree I can fix that here in this PR.

**Comment by michael-villamar** [comment]
No problem with me @aheficent

**Comment by AaronHForgeFlow** [comment]
Commit history respected now. I have not reviewed yet. But if this is green it will easier to review by others.
:smiley:

**Comment by meigallodixital** [comment]
What can I do/test/review to give a boost to merge this PR? It was blocking the rest of modules to migrate. 

Here module works perfect. I dont known if there are any functionality that we dont use that is blocke the merge.

**Comment by AaronHForgeFlow** [comment]
If you consider this is working fine you can approve this PR. Other approvals will be needed. I haven't had the time for properly review but I will as soon as possible.

**Review by meigallodixital — APPROVED** [review]
LGTM

**Review by AaronHForgeFlow — COMMENTED** [review]
For manual payments we should be able to select the OU in the payment as long as we have access to the OU. This is not happening:
image

The reports and the invoices seems correct to me.

**Comment by AaronHForgeFlow** [comment]
There are conflicts in the branch :crying_cat_face: It seems this is the commit: b6ce928dbcdd9dd706dcd86bb78babe7aeb70f88 I think if you remove this commit everything will be ok.

**Comment by meigallodixital** [comment]
I will try to prove it this afternoon and check manual payments screen (i see that comes tagged as read-only). 

The last two days I could not be aware of the changes in the repository, I'm sorry: \

I will edit this message to not create noise in the thread :)

File: account_operating_unit/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo.tools.translate import _
+from odoo import api, fields, models
```
Comment (nikul-serpentcs) [inline account_operating_unit/models/account_move.py:4]: @michael-villamar IMO translate ('_') directly use here.

**Comment by richard-willdooit** [comment]
@meigallodixital Are we going to see this merged soon?

File: account_operating_unit/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                                        operating_unit_default_get(self._uid))
+
+    @api.model
+    def create(self, vals):
+        if vals.get('move_id', False):
+            move = self.env['account.move'].browse(vals['move_id'])
+            if move.operating_unit_id:
+                vals['operating_unit_id'] = move.operating_unit_id.id
+        _super = super(AccountMoveLine, self)
+        return _super.create(vals)
```
Comment (nikul-serpentcs) [inline account_operating_unit/models/account_move.py:23]: @michael-villamar IMHO directly use `return super(AccountMoveLine, self).create(vals)`

File: account_operating_unit/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            'move_id': move.id,
+            'journal_id': move.journal_id.id,
+            'date': move.date,
+            'operating_unit_id': ou_id,
+            'account_id': move.company_id.inter_ou_clearing_account_id.id
+        }
+
+        if ou_balances[ou_id] < 0.0:
+            res['debit'] = abs(ou_balances[ou_id])
+
```
Comment (nikul-serpentcs) [inline account_operating_unit/models/account_move.py:82]: @michael-villamar IMHO Remove blank line

**Comment by meigallodixital** [comment]
> 
> 
> @meigallodixital Are we going to see this merged soon?

Now I am very busy with another project here at work, I hope to be able to resume this in a week :(

File: account_operating_unit/models/account_journal.py [hunk 4]
```diff
@@ -1,7 +1,5 @@
-# -*- coding: utf-8 -*-
-# © 2016-17 Eficent Business and IT Consulting Services S.L.
-# © 2016 Serpent Consulting Services Pvt. Ltd.
```
Comment (grindtildeath) [inline account_operating_unit/models/account_journal.py]: Please do not remove headers from previous contributors in files including code. 
Only the first line `# -*- coding: utf-8 -*-` should be removed.

**Review by grindtildeath — COMMENTED** [review]
Thanks for the port. I tested and seems to work fine.
However, the commit history is a bit messy with the change explained by @richard-willdooit on https://github.com/OCA/operating-unit/commit/fd2fc49de45f64774bf14f7cb7df026d5006f24a. So IMO would be great to have a specific commit for that (after the migration one) so it can be backported to 10.0 if needed.
Because as of now we have :
1. fd0433ec2a5eea9ac37902e53c0d37a7ccaf0862 (from 10.0 https://github.com/OCA/operating-unit/commit/fd2fc49de45f64774bf14f7cb7df026d5006f24a) which makes a change validated on #113, so this one should stay.
2. https://github.com/OCA/operating-unit/pull/114/commits/b9c083ce807f0e687f40afa32dd1064a8ee9e2ab is the migration commit, but it reverts the change without any explanation :-1: 
3. https://github.com/OCA/operating-unit/pull/114/commits/921aa2c9bf87ff920bfa3debfd1b3ef534d059a5 reverts the previous reveral
4. https://github.com/OCA/operating-unit/pull/114/commits/42ebcc6efc35989249a3403c59f15201bde8a856 reverts the reverts which reverts the previous commit...

So please remove this change from the migration commit and squash 3 and 4 together with a self explaining commit message about the issue (BTW adding a unittest covering the change in the same commit would make your contribution an awesome one :wink: )

**Comment by AaronHForgeFlow** [comment]
Superseded by #140 Do you mind if I close this @michael-villamar ?

**Comment by richard-willdooit** [comment]
I am just happy to see it move again.  #140 looks a go-er for me.  I'll poke @michael-villamar but am sure he will be OK with it ;)

**Comment by emagdalenaC2i** [comment]
> Superseded by #140 Do you mind if I close this @michael-villamar ?

@pedrobaeza #140 is already merged, so I think this PR should be closed

**Pull request closed without merging**